mh/session.py

```diff
@@ -303,6 +327,17 @@ def get_layer_players(self):
         else:
             assert False, "Can't find layer"
 
+    def attempt_leave_all_layers(self, shutdown_type):
+        assert 1 <= shutdown_type <= 2, "Invalid shutdown type"
+        if shutdown_type != 1:
+            return  # only shutdown_type 1 requires removal from layers
```

Comment:
Please add a comment to explain the differences between `layer_end` and this function including which particular issues/cases it addresses. I want to avoid repeating the same mistakes ><'.

Could you please move the `assert` and `shutdown_type` check and perform them in `recvReqShut` instead. That will get rid of the parameter.